plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeConnectorSmokeTest.java

```diff
@@ -2249,6 +2255,55 @@ public void testPartitionFilterIncluded()
         }
     }
 
+    @Test(dataProviderClass = DataProviders.class, dataProvider = "trueFalse", invocationCount = 3)
+    public void testConcurrentInsertsReconciliation(boolean partitioned)
+            throws Exception
+    {
+        int threads = 3;
+
+        CyclicBarrier barrier = new CyclicBarrier(threads);
+        ExecutorService executor = newFixedThreadPool(threads);
+        String tableName = "test_concurrent_inserts_table_" + randomNameSuffix();
+
+        assertUpdate("CREATE TABLE " + tableName + " (a INT, part INT) " +
+                (partitioned ? " WITH (partitioned_by = ARRAY['part'])" : ""));
+
+        try {
+            // insert data concurrently
+            executor.invokeAll(ImmutableList.<Callable<Void>>builder()
+                            .add(() -> {
+                                barrier.await(20, SECONDS);
+                                getQueryRunner().execute("INSERT INTO " + tableName + " VALUES (1, 10)");
```

Comment:
What should be outcome of 3 concurrent inserts like:
`INSERT INTO " + tableName + " (a, part) SELECT count(*), 10 FROM " + tableName`

Reply:
The writes to the Delta Lake table should be serializable meaning that concurrent statements like the ones above can be commited only when they don't have a [write–write conflict](https://en.wikipedia.org/wiki/Write%E2%80%93write_conflict).

So, normally, only one of the queries will succeed, while the other queries will likely fail because another version of the table exists already at the time when they are trying to write the transaction log file.